File: src/frontend/src/services/GenreService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/comiccity/genres';

export const create = (genre) => {
    return axios.post(`${API_URL}/create`, genre);
};

export const geGenre = (id) => {
    return axios.get(`${API_URL}/read/${id}`);
};

export const update = (genre) => {
    return axios.post(`${API_URL}/update`, genre);
};

export const deleteGenre = (id) => {
    return axios.delete(`${API_URL}/delete/${id}`);
};
export const getAllGenre = () => {
    return axios.get(`${API_URL}/getAll`);
};

File: src/main/java/za/ac/cput/service/genreService/GenreService.java
package za.ac.cput.service.genreService;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import za.ac.cput.domain.Genre;
import za.ac.cput.repository.GenreRepository;

import java.util.List;


@Service
public class GenreService implements IGenreService{

    @Autowired
    GenreRepository genreRepository;

    @Override
    public Genre create(Genre genre) {
        return genreRepository.save(genre);
    }

    @Override
    public Genre read(Long aLong) {
        return read(aLong);
    }


    @Override
    public Genre update(Genre genre) {
        return genreRepository.save(genre);
    }

    @Override
    public boolean delete(Long aLong) {
        genreRepository.deleteById(aLong);
        return !genreRepository.existsById(aLong);
    }

    @Override
    public List<Genre> getAll() {
        return genreRepository.findAll();
    }
}
